darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1251,110 +1208,64 @@ def load_from_checkpoint(
+            logger,
+        )
 
-        # if file_name is none, find most recent file in savepath that is a checkpoint
+        model = TorchForecastingModel.load_model(base_model_path)
+
+        # load pytorch lightning module from checkpoint
+        # if file_name is None, find most recent file in savepath that is a checkpoint
         if file_name is None:
-            path = os.path.join(
-                checkpoint_dir, "model_best_*" if best else "checkpoint_*"
-            )
-            checklist = glob(path)
-            if len(checklist) == 0:
-                raise_log(
-                    FileNotFoundError(
-                        "There is no file matching prefix {} in {}".format(
-                            "model_best_*" if best else "checkpoint_*", checkpoint_dir
-                        )
-                    ),
-                    logger,
-                )
-            file_name = max(
-                checklist, key=os.path.getctime
-            )  # latest file TODO: check case where no files match
-            file_name = os.path.basename(file_name)
+            file_name = _get_checkpoint_fname(work_dir, model_name, best=best)
 
         file_path = os.path.join(checkpoint_dir, file_name)
         logger.info("loading {}".format(file_name))
-        return TorchForecastingModel.load_model(file_path)
-
-    def _get_best_torch_device(self):
-        is_cuda = torch.cuda.is_available()
-        if is_cuda:
-            return torch.device("cuda:0")
-        else:
-            return torch.device("cpu")
-
-    def _get_learning_rate(self):
-        for p in self.optimizer.param_groups:
-            return p["lr"]
 
+        model.model = model.model.load_from_checkpoint(file_path)
+        model.load_ckpt_path = file_path
+        return model
 
-class TorchParametricProbabilisticForecastingModel(TorchForecastingModel, ABC):
-    def __init__(self, likelihood: Optional[Likelihood] = None, **kwargs):
-        """Pytorch Parametric Probabilistic Forecasting Model.
-
-        This is a base class for pytroch parametric probabilistic models. "Parametric"
-        means that these models are based on some predefined parametric distribution, say Gaussian.
-        Make sure that subclasses contain the *likelihood* parameter in __init__ method
-        and it is passed to the superclass via calling super().__init__. If the likelihood is not
-        provided, the model is considered as deterministic.
-
-        All TorchParametricProbabilisticForecastingModel's must produce outputs of shape
-        (batch_size, n_timesteps, n_components, n_params). I.e., there's an extra dimension
-        to store the distribution's parameters.
-
-        Parameters
-        ----------
-        likelihood
-            The likelihood model to be used for probabilistic forecasts.
-        """
-        super().__init__(**kwargs)
-        self.likelihood = likelihood
+    @property
+    def epochs_trained(self) -> int:
```

Comment:
Are these 4 properties used?

Reply:
All but `loss_fn` (now removed). They are used for direct "communication" between `ForecastingModel` and `PLForecastingModule`. For example likelihood is required for `_is_probabilistic()`.